### DISCUSSION TITLE
No available bundle exports error in the example tycho-its/projects/mixed.reactor/ when imported into Eclipse IDE after Cleaning

### DISCUSSION THREAD

**Discussion by markwelarz** [eclipse-tycho/tycho#2180] [Q&A]
Hello,

My mixed-reactor project has a workaround that uses the Felix Maven plugin to generate manifest files into the root of the project although not ideal this works well within the IDE.  I read the article at https://xn--lubisoft-0za.gmbh/en/articles/mixed-reactor-builds-with-tycho/ with great interest that this is no longer necessary.  Unfortunately when I follow the steps, Eclipse is showing the error `No available bundle exports 'xyz'` . I have reproduced the same behaviour with the mixed.reactor example in the Tycho repo.

Steps to reproduce:
1. Checkout the tycho-3.0.3 tag, and import the 4 projects from within mixed.reactor into Eclipse using "Import/ Maven / Existing projects"
2. Change the tycho-version property from 2.7.0 to 3.0.3 in root pom.xml file
3. Add this section to the felix.bundle/pom.xml plugin, to generate a manifest on rebuilding in Eclipse
``` 
<execution>
         <id>bundle-manifest</id>
         <phase>process-classes</phase>
	 <goals>
		 <goal>manifest</goal>
	</goals>
</execution>
```

5. Choose Maven / Update Projects, then Rebuild-all
_No errors at this point_
7. Select Project -> Clean
In problems there is the entry `No available bundle exports package felix.bundle.exported`

Cleaning/rebuilding/restarting never clears this error.  The only thing that does, is in tycho.bundle/META-INF/MANIFEST.MF is to remove the felix.bundle.exported entry and re-add it.  Using Project -> Clean will then trigger the error again.

Am I doing something wrong?  I'm using "Eclipse IDE for RCP and RAP Developer 2022-12", have installed no extra plugins and using the bundled JDK with Eclipse.

Thank you and apologies in advance for wasting anyone's time.

**Comment by laeubi** [comment]
The most important part would be if the maven build succeeds (because that actually the "Tycho Part"), the other is the IDE/PDE part for this to work, you should first check if your bundle is shown in the "Target Platform State" (you probably need to show that view explicitly), if that is missing, check if/where the manifest file is generated, if it is placed, e.g. in the target/classes/META-INF folder you need to configure PDE to find that file this is done in the file `.settings/org.eclipse.pde.core.prefs` and then `BUNDLE_ROOT_PATH= target/classes/` (as an example!).

Normally this should be done by m2e automatically, so you can try to remove you project from the workspace and make sure you have deleted all "dot files" (.project, .classpapath . settings ...) and then import as maven project again.

If you still see issues, you might report this to the [PDE project](https://github.com/eclipse-pde/eclipse.pde) where this is handeled in the IDE, there are knwon issues where one needs to restart the IDE sometimes if PDE gets stuck somewhere.

**Reply by markwelarz** [reply]
It does show in the Target Platform State (both when the error is showing and not).

Similarly, when the error is showing the manifest is in target/classes/META-INF and is correct. The /.settings/org.eclipse.pde.core.prefs is also contains `BUNDLE_ROOT_PATH= target/classes/`

Also, restarting the IDE doesn't resolve the issue either.

However, I have just discovered that if I choose Plug-in Tools / Update Classpath on the offending felix.bundle project, this seems to jolt PDE into resolving the errors.  I guess that's triggering the same thing as Remove/Add the dependency.

I will search the PDE issues and discussion to see if this is already known before I create a new issue.

Thank you so much for your help :)
